main.md

```diff
@@ -522,32 +522,25 @@ Later that same client makes a refresh request for `read` access:
+i.e. it is the recommended way to request issuance of audience restricted access tokens.
 
-The `resource` token request parameter as defined in [@!RFC8707] MAY be used in the token request to request the creation of an audience restricted access token (as recommended in [@I-D.ietf-oauth-security-topics]). If the client uses this parameter, the AS MUST consider the audience restriction defined by the `locations` elements of the `authorization_details` to filter the authorization data objects applicable to the respective resource(s). 
+For our running example, the client MAY ask for all permissions of the approved grant of type `payment_iniation` applicable to the resource server residing at `https://example.com/payments` as follows:  
 
-The logic is as follows:
-
-* For every authorization details object without a `locations` element: the authorization server treats it as applicable to all resources, i.e. it assigns this authorization details object to the access token. 
-* For every authorization details object with a `locations` element: the authorization server adds this object to the access token, if at least one of the `locations` values exactly matches the `resource` token request parameter value. The authorization server MUST compare both values using an exact byte match of the string values.
-
-For example, the following token request selects authorization details applicable for the resource server represented by the URI `https://example.com/payments`.
-
-```http
-POST /token HTTP/1.1
-Host: as.example.com
-Authorization: Basic czZCaGRSa3F0MzpnWDFmQmF0M2JW
-Content-Type: application/x-www-form-urlencoded
+```JSON
+[
+   {
+      "type": "payment_initiation",
+      "locations": [
+         "https://example.com/payments"
+      ]
+   }
+]
 
-grant_type=authorization_code&code=SplxlOBeZQQYbYS6WxSbIA
-&redirect_uri=https%3A%2F%2Fclient%2Eexample%2Ecom%2Fcb
-&resource=https%3A%2F%2Fexample%2Ecom%2Fpayments
 ```
 
-Using the example given above, this request would result in the assignment of the `payment_initiation` authorization details object from (#authz_details) to the access token to be issued (see below).
-
 # Token Response
-The authorization details assigned to the access token issued in a token response are determined by the `authorization_detail` parameter of the corresponding token request as well as any related parameters such as `resource` and `scope`.  If the client does not specify any of those token request parameters, the AS determines the resulting authorization details at its discretion.
+The authorization details assigned to the access token issued in a token response are determined by the `authorization_detail` parameter of the corresponding token request. If the client does not specify the `authorization_detail` token request parameters, the AS determines the resulting authorization details at its discretion.
```

Comment:
I don't want to lose the fact that `scope` and `resource` exist at the same top level, and we don't want to imply that they're mutually exclusive. I also think that it's up to the AS to figure out how those go together.

```suggestion
The authorization details assigned to the access token issued in a token response are determined by the `authorization_detail` parameter of the corresponding token request. If the client does not specify the `authorization_detail` token request parameters, the AS determines the resulting authorization details at its discretion. The authorization server MAY consider the values of other parameters such as `resource` and `scope` during this processing, and the details of such considerations are outside the scope of this specification.
```